From 8c2ff743f25a5a690115d866c409d1a971555c06 Mon Sep 17 00:00:00 2001
From: Sebastian Niciarz <sebastian.niciarz@intel.com>
Date: Thu, 18 Sep 2014 10:00:26 +0200
Subject: [PATCH 2/4] pn544_nfc: rename xfer platform data to i2c_mtu and make
 it optional

i2c_mtu should be optional with fallback to 'no limit'

Change-Id: I3c4617ef77aa6ac95cf79ef38414376137c5b91a
Signed-off-by: Sebastian Niciarz <sebastian.niciarz@intel.com>
---
 drivers/staging/pn544/pn544.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/drivers/staging/pn544/pn544.c b/drivers/staging/pn544/pn544.c
index 1cdcfcc6f247..bad742ebe1c4 100644
--- a/drivers/staging/pn544/pn544.c
+++ b/drivers/staging/pn544/pn544.c
@@ -435,14 +435,10 @@ static int pn544_acpi_probe(struct i2c_client *client,
 		pn544_dev->firm_gpio = gpio;
 	}
 
-	status = acpi_evaluate_integer(ACPI_HANDLE(dev), "XFER",
+	status = acpi_evaluate_integer(ACPI_HANDLE(dev), "I2C_MTU",
 				NULL, &data);
-	if (ACPI_FAILURE(status)) {
-		pr_err("Error evaluating ACPI XFER object");
-		return -ENODEV;
-	}
-
-	pn544_dev->max_i2c_xfer_size = data;
+	if (ACPI_SUCCESS(status))
+		pn544_dev->max_i2c_xfer_size = data;
 
 	return 0;
 }
-- 
1.9.1

